vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/WindowSteps.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.steps.ui.web;
+
+import java.awt.GraphicsEnvironment;
+
+import org.jbehave.core.annotations.When;
+import org.openqa.selenium.Dimension;
+import org.vividus.monitor.TakeScreenshotOnFailure;
+import org.vividus.selenium.BrowserWindowSize;
+import org.vividus.selenium.IBrowserWindowSizeProvider;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.manager.IWebDriverManager;
+import org.vividus.softassert.ISoftAssert;
+
+@TakeScreenshotOnFailure
+public class WindowSteps
+{
+    private final IWebDriverManager webDriverManager;
+    private final IWebDriverProvider webDriverProvider;
+    private final IBrowserWindowSizeProvider browserWindowSizeProvider;
+    private final ISoftAssert softAssert;
+
+    public WindowSteps(IWebDriverManager webDriverManager, IWebDriverProvider webDriverProvider,
+                       IBrowserWindowSizeProvider browserWindowSizeProvider, ISoftAssert softAssert)
+    {
+        this.webDriverManager = webDriverManager;
+        this.webDriverProvider = webDriverProvider;
+        this.browserWindowSizeProvider = browserWindowSizeProvider;
+        this.softAssert = softAssert;
+    }
+
+    /**
+     * Change current browser window size to the specified one.
+     * @param targetSize Any browser window size in pixels, e.g. 800x600.
+     *                    Note: browser window size should be smaller than the current screen resolution.
+     */
+    @When("I change current window size to `$targetSize`")
+    public void resizeCurrentWindow(BrowserWindowSize targetSize)
+    {
+        boolean remoteExecution = webDriverProvider.isRemoteExecution();
+        Dimension screenSize = !remoteExecution && GraphicsEnvironment.isHeadless()
```

Comment:
```suggestion
        boolean localBrowser = !webDriverProvider.isRemoteExecution();
        Dimension screenSize = localBrowser && GraphicsEnvironment.isHeadless()
```

Reply:
remoteExecution variable is also used at line 58. I'll better leave it as is so as not to call webDriverProvider.isRemoteExecution() twice.